docs/source/how_to_guides/synthetic_nlp.ipynb

```diff
@@ -0,0 +1,607 @@
+                "We can also visualize the sample(s) by doing pythonic or NumPy indexing on a `StreamingDataset`."
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "# Fetch the 10th sample and print it on a console\n",
+                "print(f'Sample 10: {train_dataset[10]}')\n",
+                "\n",
+                "# Fetch multiple samples\n",
+                "indices = [-1, 30, [12, -14], slice(-1, -10, -2), np.array([10, -20])]\n",
+                "for indx in indices:\n",
+                "    print(f'Sample {indx}: {train_dataset[indx]}')"
+            ]
+        },
+        {
+            "attachments": {},
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "Below are some utility methods about the dataset which would be highly useful for debugging and model training. For more information on the `StreamingDataset` parameters, check out the [API reference](https://streaming.docs.mosaicml.com/en/stable/api_reference/generated/streaming.StreamingDataset.html#streaming.StreamingDataset)."
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "# Get the total number of samples\n",
+                "print(f'Total number of samples: {train_dataset.num_samples}')\n",
+                "\n",
+                "# Get the number of shard files\n",
+                "print(f'Total number of shards: {len(train_dataset.shards)}')\n",
+                "\n",
+                "# Get the number of samples inside each shard files.\n",
+                "# Number of samples in each shard can vary based on each sample size.\n",
+                "print(f'Number of samples inside each shards: {train_dataset.samples_per_shard}')"
+            ]
+        },
+        {
+            "attachments": {},
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "we can now wrap our streaming datasets in a standard PyTorch dataloaders for training!"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "train_dataloader = DataLoader(train_dataset, batch_size=batch_size)\n",
+                "val_dataloader = DataLoader(val_dataset, batch_size=batch_size)"
+            ]
+        },
+        {
+            "attachments": {},
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "The `StreamingDataset` class supports a `batch_size` parameter, if when provided by the user, worker indices will be constructed so that there \n",
```

Comment:
Is batch_size becoming a required parameter? Shall I remove this section?

Reply:
Oh yeah it's required. I'll edit and say it's required

Reply:
edited